net.rocketpool/image/src/app.ts

```diff
@@ -19,145 +23,119 @@ interface Event {
   logIndex: number;
 }
 
+type Provider = ethers.providers.JsonRpcProvider | ethers.providers.WebSocketProvider;
+
 /////////////////////////////////////////////////////////////////////////////////////////
 
 async function readEventsFromETH(
-  startBlock,
-  endBlock,
-  clb: (_: Event) => boolean,
-  stride = 10000
-) {
+  rethContract: RethAbi,
+  startBlock: number,
+  endBlock: number,
+  clb: (_: EventRow) => boolean,
+): Promise<void> {
   console.log(
-    "Considering events in block range [%s, %s] (%s blocks)",
-    startBlock,
-    endBlock,
-    endBlock - startBlock
+    `Considering events in block range [${startBlock}, ${endBlock}] (${endBlock - startBlock})`,
   );
 
-  let fromBlock = startBlock;
-  let toBlock = fromBlock;
-
-  do {
-    fromBlock = toBlock + 1;
-    toBlock = Math.min(endBlock, fromBlock + stride);
+  const eventFilters = [
+    rethContract.filters.TokensMinted(),
+    rethContract.filters.TokensBurned()
+  ] as const;
+  const eventsRowsPromises = eventFilters.map(async (eventFilter) => {
+    const events = await rethContract.queryFilter(eventFilter, startBlock, endBlock);
+
+    return events.map(async (event) => {
+      const block = await rethContract.provider.getBlock(event.blockNumber);
```

Comment:
Why not use `event.getBlock()` here?

https://github.com/ethers-io/ethers.js/blob/master/packages/contracts/src.ts/index.ts#L96C5-L96C13

Reply:
Yes, that is a good catch -- will use this shortcut!
```js
event.getBlock = () => {
    return contract.provider.getBlock(receipt.blockHash);
};
```
(c) https://github.com/ethers-io/ethers.js/blob/master/packages/contracts/lib.esm/index.js#L256C1-L258C19